src/Aspire.Hosting.Azure.Functions/DurableTask/DurableTaskSchedulerExtensions.cs

```diff
@@ -0,0 +1,450 @@
+    /// </summary>
+    /// <param name="builder">The builder for the distributed application.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="configure">(Optional) Callback that exposes the resource allowing for customization.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{DurableTaskSchedulerResource}" />.</returns>
+    public static IResourceBuilder<DurableTaskSchedulerResource> AddDurableTaskScheduler(this IDistributedApplicationBuilder builder, [ResourceName] string name, Action<IResourceBuilder<DurableTaskSchedulerResource>>? configure = null)
+    {
+        DurableTaskSchedulerResource resource = new(name);
+
+        var resourceBuilder = builder.AddResource(resource);
+
+        configure?.Invoke(resourceBuilder);
+
+        return resourceBuilder;
+    }
+
+    /// <summary>
+    /// Configures a Durable Task Scheduler resource to be emulated.
+    /// </summary>
+    /// <param name="builder">The Durable Task Scheduler resource builder.</param>
+    /// <param name="configureContainer">Callback that exposes the underlying container used for emulation allowing for customization.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{DurableTaskSchedulerResource}" />.</returns>
+    /// <remarks>
+    /// This version of the package defaults to the <inheritdoc cref="DurableTaskConstants.Scheduler.Emulator.Container.Tag" /> tag of the <inheritdoc cref="DurableTaskConstants.Scheduler.Emulator.Container.Image" /> container image in the <inheritdoc cref="DurableTaskConstants.Scheduler.Emulator.Container.Registry" /> registry.
+    /// </remarks>
+    /// <example>
+    /// The following example creates a Durable Task Scheduler resource that runs locally in an emulator and referencing that resource in a .NET project.
+    /// <code lang="csharp">
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// var scheduler = builder.AddDurableTaskScheduler("scheduler")
+    ///                        .RunAsEmulator();
+    ///
+    /// var taskHub = scheduler.AddDurableTaskHub("taskhub");
+    ///
+    /// builder.AddProject&lt;Projects.MyApp&gt;("myapp")
+    ///        .WithReference(taskHub);
+    /// 
+    /// builder.Build().Run();
+    /// </code>
+    /// </example>
+    public static IResourceBuilder<DurableTaskSchedulerResource> RunAsEmulator(this IResourceBuilder<DurableTaskSchedulerResource> builder, Action<IResourceBuilder<DurableTaskSchedulerEmulatorResource>>? configureContainer = null)
+    {
+        if (builder.ApplicationBuilder.ExecutionContext.IsPublishMode)
+        {
+            return builder;
+        }
+
+        builder
+            .WithEndpoint(name: DurableTaskConstants.Scheduler.Emulator.Endpoints.Worker, scheme: "http", targetPort: 8080)
+            .WithEndpoint(name: DurableTaskConstants.Scheduler.Emulator.Endpoints.Dashboard, scheme: "http", targetPort: 8082)
+            .WithAnnotation(
```

Comment:
WithEnvironment?

Reply:
Updated that, too.  I suppose health checks for the emulator could be useful, though I need to look at the API and examples.  I'd be inclined to leave that as a separate, follow up PR.